torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -420,6 +450,35 @@ void ProcessGroupAgent::enqueueRecv(RecvWork work) {
       std::move(work)));
 }
 
+void ProcessGroupAgent::markFutureWithMessage(Message& message) {
+  auto id = message.id();
+  std::shared_ptr<FutureMessage> fm = nullptr;
+  {
+    std::lock_guard<std::mutex> lock{futureMutex_};
+    const auto& futureInfo = futures_.find(id);
```

Comment:
can you check futures_.find(id) != futures_end()? Maybe early exit in this case is ok.